packages/parser/src/components/parser/statement/functionStatement.ts

```diff
@@ -0,0 +1,54 @@
+import Statement from ".";
+
+import { TokenTypes } from "../../../constants/bhaiLangSpec";
+import { NodeType } from "../../../constants/constants";
+import TokenExecutor from "../tokenExecutor";
+import { ASTNode } from "../types/nodeTypes";
+
+import Expression from "./expression";
+import NullLiteral from "./expression/literals/nullLiteral";
+
+export default class FunctionStatement extends Statement {
+  _nullLiteral: NullLiteral;
+
+  constructor(tokenExecutor: TokenExecutor, nullLiteral: NullLiteral) {
+    super(tokenExecutor);
+    this._nullLiteral = nullLiteral;
+  }
+
+  getStatement(): ASTNode {
+    this._tokenExecutor.eatTokenAndForwardLookahead(
+      TokenTypes.FUNDA_TYPE
+    );
+
+    const declaration = this._getFunctionDeclaration();
+
+    // this._tokenExecutor.eatTokenAndForwardLookahead(TokenTypes.SEMI_COLON_TYPE);
+
+    return {
+      type: NodeType.FunctionStatement,
+      declaration,
+    };
+  }
+
+
+  private _getFunctionDeclaration(): ASTNode {
+    const id = Expression.getExpressionImpl(
```

Comment:
`nit`: Change the name of this const from id to functionSignature or something else which actually tells what this is. Id is generic for identifier only.

Reply:
updated code